File: Model/Card.swift
import Foundation

// Define an enum Suit with cases: spades, hearts, diamonds, clubs.
enum Suit {
    case spades, hearts, diamonds, clubs
}

// Define an enum Rank with cases for Ace, 2 through 10, Jack, Queen, King.
// Make it CaseIterable to easily create a full deck.
// Consider giving ranks appropriate integer raw values (e.g., Ace = 1, King = 13)
// if that might be useful for game logic later, but simple cases are fine for now.
enum Rank: Int, CaseIterable {
    case ace = 1
    case two, three, four, five, six, seven, eight, nine, ten
    case jack, queen, king
}

// Define a struct Card with two properties:
// suit: Suit
// rank: Rank
// Make it Identifiable and Equatable for UI and comparison purposes.
struct Card: Identifiable, Equatable {
    let id = UUID() // Identifiable conformance
    let suit: Suit
    let rank: Rank

    // Equatable conformance (Swift can synthesize this for structs whose members are Equatable,
    // but it's good to be explicit or know how to do it if needed)
    // static func == (lhs: Card, rhs: Card) -> Bool {
    //     return lhs.suit == rhs.suit && lhs.rank == rhs.rank
    // }
}

File: BhabhiGameAndroid/app/src/main/java/com/example/bhabhigameandroid/Card.kt
package com.example.bhabhigameandroid

import java.util.UUID

// Define an enum Suit
enum class Suit {
    SPADES, HEARTS, DIAMONDS, CLUBS
}

// Define an enum Rank with integer values
enum class Rank(val value: Int) {
    TWO(2), THREE(3), FOUR(4), FIVE(5), SIX(6), SEVEN(7), EIGHT(8), NINE(9), TEN(10),
    JACK(11), QUEEN(12), KING(13), ACE(14) // Ace high
}

// Define a data class Card
data class Card(
    val suit: Suit,
    val rank: Rank,
    val id: String = "${rank.name}_OF_${suit.name}", // Unique ID based on rank and suit
    var isPlayed: Boolean = false
) : Comparable<Card> {
    override fun compareTo(other: Card): Int {
        // Primary sort by rank value, secondary by suit (e.g., Spades > Hearts > Diamonds > Clubs)
        // This order is arbitrary for suits but provides a consistent sort.
        if (this.rank.value != other.rank.value) {
            return this.rank.value.compareTo(other.rank.value)
        }
        return this.suit.compareTo(other.suit) // Enum compareTo is by ordinal
    }
}
